### PR TITLE
Resolve issues with Notebook Handler

### LINKED ISSUE #122: Error when closing a console
Here are the steps to reproduce:

1. Start a new console
2. Switch focus to a notebook, then to the console, then back to the notebook
3. Close the console

console-current-widget-error

### LINKED ISSUE #133: Adding breakpoints Bug
chrome-capture (19)

Scenerio -> 

Set breakpoints in notebook1 close that notebook1. Go to notebook2

### PR DESCRIPTION
**Pull request by KsavinN**
Fixes #133 
Fixes #122 

and resolve the 
console_breakpoint_hover_issue
additional issue:

### CODE REVIEW

**Comment by jtpio** [comment]
The diff contains changes from other (merged) pull requests (stop button, tests).

File: src/handlers/cell.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export class CellManager implements IDisposable {
   }
 
   isDisposed: boolean;
+  iteration: number = 0;
```
Comment (KsavinN) [inline src/handlers/cell.ts]: I will removed that. That was used for investigation of bugs

File: src/handlers/notebook.ts [hunk 2]
```diff
@@ -46,30 +47,36 @@ export class NotebookHandler implements IDisposable {
...
     this.isDisposed = true;
     this.cellManager.dispose();
-    this.notebookTracker.activeCellChanged.disconnect(
-      this.onActiveCellChanged,
-      this
-    );
+    // this.notebookTracker.activeCellChanged.disconnect(
+    //   this.onActiveCellChanged,
+    //   this
+    // );
```
Comment (KsavinN) [inline src/handlers/notebook.ts]: I'm considering to remove this cause we have Signal.clear which should remove all signals in that instance.

File: src/handlers/cell.ts [hunk 3]
```diff
@@ -165,6 +169,9 @@ export class CellManager implements IDisposable {
   }
 
   protected removeListener(cell: CodeCell) {
+    if (cell.disposed) {
```
Comment (jtpio) [inline src/handlers/cell.ts]: This should be `cell.isDisposed` to test if a cell is disposed.

**Comment by jtpio** [comment]
I left a `TODO` to check whether we could remove the call to `requestAnimationFrame`.

**Pull request merged**